src/clients/QBittorrent.ts

```diff
@@ -215,76 +221,140 @@ export default class QBittorrent implements TorrentClient {
 	}
 
 	async isInfoHashInClient(infoHash: string): Promise<boolean> {
-		const responseText = await this.request(
-			"/torrents/properties",
-			`hash=${infoHash}`,
+		const torrent = await this.getTorrentInfo(infoHash);
+		return torrent.length > 0;
+	}
+
+	async addTorrent(formData: FormData): Promise<void> {
+		await this.request("/torrents/add", formData);
+	}
+
+	async recheckTorrent(infoHash: string): Promise<void> {
+		const torrent = await this.getTorrentInfo(infoHash);
+		if (torrent.length === 0) {
+			throw new Error("Torrent not found in client");
+		}
+		await this.request(
+			"/torrents/recheck",
+			`hashes=${torrent[0].hash}`,
 			X_WWW_FORM_URLENCODED,
 		);
-		try {
-			const properties = JSON.parse(responseText);
-			return properties && typeof properties === "object";
-		} catch (e) {
-			return false;
-		}
 	}
+
+	/*
+	@param searchee the Searchee we are generating off (in client)
+	@return either a string containing the path or a error mesage
+	*/
 	async getDownloadDir(
-		searchee: Searchee,
+		searchee: SearcheeWithInfoHash,
 	): Promise<
 		Result<string, "NOT_FOUND" | "TORRENT_NOT_COMPLETE" | "UNKNOWN_ERROR">
 	> {
-		let torrentInfo: TorrentConfiguration;
 		try {
-			if (await this.isInfoHashInClient(searchee.infoHash!)) {
-				torrentInfo = await this.getTorrentConfiguration(searchee);
-				if (torrentInfo.save_path === undefined) {
-					return resultOfErr("NOT_FOUND");
-				}
+			const result = await this.getTorrentInfo(searchee.infoHash);
+			if (result.length === 0) {
+				return resultOfErr("NOT_FOUND");
 			}
+			const torrentInfo = result[0];
+			if (!this.isTorrentComplete(torrentInfo.state)) {
+				return resultOfErr("TORRENT_NOT_COMPLETE");
+			}
+			const savePath = await this.generateCorrectSavePath(
+				searchee,
+				torrentInfo,
+			);
+			return resultOf(savePath);
 		} catch (e) {
 			if (e.message.includes("retrieve")) {
 				return resultOfErr("NOT_FOUND");
 			}
 			return resultOfErr("UNKNOWN_ERROR");
 		}
-		return resultOf(torrentInfo!.save_path);
 	}
 
-	async getTorrentConfiguration(
+	/*
+	@param searchee the Searchee we are generating off (in client)
+	@param subfolderLayout whether it is subfolder layout or not
+	@return string absolute location from client with content layout considered
+	 */
+	async generateCorrectSavePath(
```

Comment:
looks like this one doesn't need to be async either. also, a naming nit - maybe `getCorrectSavePath`? we're not really generating/producing anything here